00-introduction.adoc

```diff
@@ -1,199 +1,265 @@
-consider its method names and arguments to be an API.
-
-Essentially it’s all the same concept, but when talking about Web APIs
-we’re talking about the utilization of network protocols such as HTTP,
-AMQP, etc., and instead of command-line arguments or method parameters,
-we’re working with URLs, query string parameters, or chunks of data like
-JSON.
-
-The goal of a Web API is to provide other applications with access to
-specific subset of functionality and data that this application owns.
-Some APIs are public (anyone can get the weather from
-http://api.weather.gov[api.weather.gov]), some are private (it’s
-probably tough to get access to JPMorgan Chase’s Enterprise Customers
-API), and some are a mixture.
-
-APIs can give you data for a single-page JavaScript application, handle
-payment information so you don’t have to worry about storing credit card
-details, post a Facebook status on a users timeline, or share the same
-data across a myriad of different devices. Watch that Netflix show on
-your Xbox, it’s all good, APIs have you covered.
-
-For a long time my job was designing and building APIs, Then it was
-coaching and advising other teams how to build APIs. In more recent
-years, it has shifted towards helping different teams building different
-software interact with each others systems. These are sometimes
-architectural concerns, but a lot of it comes down to clients calling
-other systems and knowing what to do with the responses.
-
-Essentially the role has been making HTTP interactions reliable, warning
-people of potential problems like: don’t assume everything is going to
-be JSON, normalizing state management into the API instead of having
-loads of clients doing subtly different guesswork, how to configure
-timeouts to stop half the company’s applications crashing, and improving
-performance with smart usage of HTTP caching. This advice seemed like it
-should also go into a book, because I’ve learned a substantial amount
-from my time working at WeWork.
-
-This book aims to help you interact with an API that is already built.
-It will explain things from the users perspective, and if you do happen
-to be an API developer, this might give you some insight into how users
-will be interacting with your API. Everyone wins. That said, if you want
-help building fantastic web APIs, this might not be first book you
-should read, and instead you should consider picking up a copy of
-http://apisyouwonthate.com[_Build APIs You Won’t Hate_]. This is for the
-frontend folks, and those building systems that talk to other systems.
+An API ("Application Program Interface") is a rather generic term in the world
+of computer science and programming. A Linux command-line application would
+consider the command names and options to be an API. A Java module would also
+consider its classes, method names, and arguments to be an API. 
+
+Basically an API means code somebody else wrote, and a Web APIs is just code
```

Comment:
how about "an API allows you to send command requests to code somebody else wrote?"